sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -96,6 +98,11 @@ private[sql] class SessionState(
     hadoopConf
   }
 
+  def newHadoopConfWithCaseInsensitiveOptions(options: CaseInsensitiveStringMap): Configuration = {
```

Comment:
I don't think we should pollute `SessionState` with the case insensitive map stuff. Can we inline this method?